addons/web/static/src/views/layout.js

```diff
@@ -0,0 +1,39 @@
+/** @odoo-module **/
+
+import { ControlPanel } from "@web/search/control_panel/control_panel";
+import { SearchPanel } from "@web/search/search_panel/search_panel";
+
+const { Component } = owl;
+
+export class Layout extends Component {
+    setup() {
+        this.components = { ...this.constructor.components, ...this.props.components };
+    }
+
+    get display() {
+        if (this.props.display) {
+            return this.props.display;
+        }
+        if (this.env.searchModel) {
+            return this.env.searchModel.display;
+        }
```

Comment:
Do we need these fallbacks? In general, I don't like when there are several way to do something.

Reply:
same for components in setup

Reply:
I think the `components` case makes sense because it falls back to the generic `ControlPanel` and `SearchPanel` classes, which we don't necessarily want to specifiy on each view class definition.

As for the `display` property it is purely theoretical: I wanted the `Layout` component to be independant of the View/SearchModel ecosystem, but still rely on it when it can. We can remove this if you want, as the layout will probably only be instanciated through views and actions.